thirdeye/thirdeye-frontend/app/utils/utils.js

```diff
@@ -53,13 +53,24 @@ export function humanizeFloat(f) {
 }
 
 /**
- * Formatter for the human-readable change values in percent with 1 decimal
+ * Formatter for the human-readable change values displayed in percent with 1 decimal. Truncates extreme values > 10x.
  * @param {float} f float number
  * @return {string} human-readable change string
+ *
+ * @example
+ * 0.5177  => +51.8%
+ * -0.0001 => -0.0%
+ * 10.001  => +1000%+
+ * -10.01  => -1000%+
  */
 export function humanizeChange(f) {
   if (!isValidForDisplay(f)) { return '-'; }
-    return `${f > 0 ? '+' : ''}${Math.round(f * 100).toFixed(1)}%`;
+  if (f > 10) {
```

Comment:
nit: let's use a more meaningful name instead of `f`

Reply:
yeah. I'll do this across the entire utils in a separate PR